Extension/src/LanguageServer/configurations.ts

```diff
@@ -736,34 +737,57 @@ export class CppProperties {
-        const result: string[] = [];
-        if (paths) {
-            paths = this.resolveDefaults(paths, defaultValue);
-            paths.forEach(entry => {
-                const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
-                result.push(...entries);
-            });
-        }
+        const resolvedVariables: string[] = [];
+        if (paths === undefined) {
+            return resolvedVariables;
+        }
+        paths = this.resolveDefaults(paths, defaultValue);
+        paths.forEach(entry => {
+            const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
+            resolvedVariables.push(...entries);
+        });
         if (!glob) {
-            return result;
-        }
-
-        const globResult: string[] = [];
-        for (let res of result) {
-            // fastGlob will expand the ending double wildcard. temporary strip them before expanding
-            const recursive: boolean = res.endsWith('**');
-            if (recursive) {
-                res = res.slice(0, res.length - 2);
+            return resolvedVariables;
+        }
+        const resolvedGlob: string[] = [];
+        for (let res of resolvedVariables) {
+            let counter: number = 0;
+            let slashFound: boolean = false;
+            const lastIndex: number = res.length - 1;
+            // Detect and glob all wildcard variations by looking at last character in the path first.
+            for (let i: number = lastIndex; i >= 0; i--) {
+                if (res[i] === '*') {
+                    counter++;
+                } else if (res[i] === '/' || res[i] === '\\' && isWindows) {
+                    counter++;
+                    slashFound = true;
+                    break;
+                } else {
+                    break;
+                }
+            }
+            let suffix: string = '';
+            if (slashFound) {
+                suffix = res.slice(res.length - counter);
+                res = res.slice(0, res.length - counter);
             }
-            // fastGlob can't deal with backslash-separated path => remove them
-            const normalized: string = res.replace(/\\/g, '/');
-            const cwd: string = this.rootUri?.fsPath?.replace(/\\/g, '/') || '';
+            let normalized: string = '';
+            let cwd: string = '';
+            if (isWindows) {
+                normalized = res.replace(/\\/g, '/');
+                cwd = this.rootUri?.fsPath?.replace(/\\/g, '/') || '';
+            }
+            const isGlobPattern: boolean = res.includes('*');
+            if (isGlobPattern) {
             // fastGlob silently strip non-found paths. limit that behavior to dynamic paths only
```

Comment:
Can you re-word this existing comment to something like `// fastGlob silently strips non-found paths. Limit that behavior to dynamic paths only.` And I think it needs to be tabbed over.